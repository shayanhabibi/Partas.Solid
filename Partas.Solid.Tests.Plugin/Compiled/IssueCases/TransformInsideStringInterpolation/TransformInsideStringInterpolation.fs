module Partas.Solid.Tests.IssueCases.TransformInsideStringInterpolation

open Partas.Solid
open Fable.Core

[<Erase>]
type TestStringInterpolation() =
    interface VoidNode
    [<Erase>]
    member val index: int = unbox null with get,set
    [<SolidTypeComponentAttribute>]
    member props.constructor =
        div().style' {| paddingLeft = $"{props.index}px" |} {
            div().style' {| paddingLeft = $"{props.index}px" |} {
                div().style' {| paddingLeft = $"{props.index / 2}px" |}
            }
        } 
