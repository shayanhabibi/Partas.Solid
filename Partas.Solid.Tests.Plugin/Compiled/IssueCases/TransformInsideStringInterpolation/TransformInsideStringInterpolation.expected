import { splitProps } from "solid-js";

export function TestStringInterpolation(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["index"]);
    return <div style={{
            paddingLeft: `${PARTAS_LOCAL.index}px`,
        }}>
        <div style={{
                paddingLeft: `${PARTAS_LOCAL.index}px`,
            }}>
            <div style={{
                    paddingLeft: `${~~(PARTAS_LOCAL.index / 2)}px`,
                }} />
        </div>
    </div>;
}
